runtime/src/main/cpp/MemorySharedRefs.cpp

```diff
@@ -8,6 +8,19 @@
 #include "MemorySharedRefs.hpp"
 #include "Runtime.h"
 
+namespace {
```

Comment:
Code supporting `SharedRef` doesn't belong to this file: it is a high-level user of `MemorySharedRefs.hpp`.
I believe it should be moved to a separate file then, e.g. `SharedRef.cpp`.